MIGRATION.md

```diff
@@ -0,0 +1,76 @@
+
+Now that Manage is there, we can move storage of the policies into the Manage backend (MongoDB) and frontend, so we can make use of the existing facilities for managing the history, integrate the admin UI and enforce referential integrity.
+
+The storage in Manage is done in JSON format. Upon PUSH in Manage this is sent to the PDP-server which translates it to XACML and stores it in its local database for runtime evaluation. The PUSH in Manage is when policy changes become active, no delays anymore.
+
+## How does the migration work
+
+A new policies table is added to PDP next to the existing one. Manage can import the existing policies, process and store them and push them to the new table.
+
+The migration can be done in two stages:
+1. The new table exists besides the old one but is not used for policy evaluation. Manage can pull from the old table (as often as desired), and push back to the new table. Manage can then also show any differences between old and new, so certainty can be gained that the conversion is successful. Changing policies at this time still happens from PDP GUI.
+2. The new table is switched to production. The old table is no longer used and the PDP GUI also not.
+
+## Requirements
+
+To migrate your rules from PDP to Manage, you need:
+
+* OpenConext-pdp >= 4.1.0 (>= 5.1 preferred), << 7.0.0
+* OpenConext-manage >= 7.4.0 (>= 7.4.9 preferred), << 9.0.0
+* OpenConext-dashboard (when using), >= 12.3.17
+
+## Preparation
+
+Ensure that Manage's `application.yml` contains a PUSH section for PDP and a Spring MySQL datasouce (for comparing the result, only needed during the migration).
+
+```
+push:
+  pdp:
+    url: http://localhost:8081/pdp/api/manage/push
+    policy_url: http://localhost:8081/pdp/api/manage/policies
+    decide_url: http://localhost:8081/pdp/api/manage/decide
+    user: pdp_admin
+    name: OpenConext PDP
+    password: secret
+    enabled: true
+
+spring:
+  datasource:
+    url: jdbc:mysql://localhost:3306/pdpserver?permitMysqlScheme
+    username: root
+    password:
+    driverClassName: org.mariadb.jdbc.Driver
+```
+
+In PDP's `application.properies`, besides the API user listed above, also set pushTestMode=True:
+```
+manage.pushTestMode=True
+```
+
+## Load policies in Manage and test
+
+Now in Manage, you can ask Manage to import all policies from PDP under the Policies tab. This will report what it does and if you push it back, it also provides ways to compare. You can view the imported policies under a related entity (e.g. an SP) in the Policies tab for that entity. You can repeat the fetch and push as often as you like as long as `manage.pushTestMode=False`.
```

Comment:
You can also get an overview of all policies in the policies tab which is located at the second level (where you can find "service providers" etc